src/tools/clu_funcs.c

```diff
@@ -1154,6 +1154,80 @@ static void wolfCLU_AddNameEntry(WOLFSSL_X509_NAME* name, int type, int nid,
 }
 
 
+/* Input 'n' is a terminated string in the form of '/CN=name/C=company'
+ * returns a newly created WOLFSSL_X509_NAME on success */
+WOLFSSL_X509_NAME* wolfCLU_ParseX509NameString(char* n, int nSz)
+{
+    WOLFSSL_X509_NAME* ret = NULL;
+    char* word, *end;
+    char* deli = (char*)"/";
+    char* entry = NULL;
+    char  tag[5] = {0};
+
+    if (n == NULL || nSz <= 0) {
+        return NULL;
+    }
+
+    ret = wolfSSL_X509_NAME_new();
+    if (ret == NULL) {
+        WOLFCLU_LOG(WOLFCLU_L0, "error allocating name structure");
+        return NULL;
+    }
+
+    tag[0] = '/';
+    for (word = strtok_r(n, deli, &end); word != NULL;
+            word = strtok_r(NULL, deli, &end)) {
+        int tagSz = 0;
+        int nid;
+
+        tagSz = (int)strcspn(word, "=");
+        if (tagSz <= 0) {
+            WOLFCLU_LOG(WOLFCLU_L0, "error finding '=' char in name");
+            wolfSSL_X509_NAME_free(ret);
+            ret = NULL;
+            break;
+        }
+
+        tagSz = tagSz + 1; /* include the '=' char */
+        if (tagSz + 2 > (int)sizeof(tag)) { /* +2 for '/' and '\0' chars */
+            WOLFCLU_LOG(WOLFCLU_L0, "found a tag that was too large!");
+            wolfSSL_X509_NAME_free(ret);
+            ret = NULL;
+            break;
+        }
+        else if (tagSz + 1 > nSz) {
+            WOLFCLU_LOG(WOLFCLU_L0, "error, entry would be past buffer end");
+            wolfSSL_X509_NAME_free(ret);
+            ret = NULL;
+            break;
+        }
+        else {
+            XMEMCPY(tag + 1, word, tagSz);
+            tag[tagSz + 1] = '\0'; /* append terminating character */
+        }
+
+        if (ret != NULL) {
+            int encoding = CTC_UTF8;
```

Comment:
Move declaration to start of function.

Reply:
Had left it at the top of scope here to only throw it on the stack when the if statement is true and to limit the variables scope, there's no harm though in moving to the top of the function and it will trade off for better readability, can move it.